pkg/util/node/node.go

```diff
@@ -45,14 +45,21 @@ const (
 // GetHostname returns OS's hostname if 'hostnameOverride' is empty; otherwise, return 'hostnameOverride'.
 func GetHostname(hostnameOverride string) string {
 	hostname := hostnameOverride
-	if hostname == "" {
+	if len(hostname) == 0 {
 		nodename, err := os.Hostname()
 		if err != nil {
 			glog.Fatalf("Couldn't determine hostname: %v", err)
 		}
 		hostname = nodename
 	}
-	return strings.ToLower(strings.TrimSpace(hostname))
+
+	// Trim whitespaces first to avoid getting an empty hostname
+	// For linux, the hostname is read from file /proc/sys/kernel/hostname directly
+	hostname = strings.TrimSpace(hostname)
+	if len(hostname) == 0 {
+		glog.Fatalf("Empty hostname is invalid")
```

Comment:
I don't like the use of `glog.Fatalf` here. It calls `os.Exit` and I perceive this as a bad thing. It does not allow for any error handling or recovery here. A panic or returning an error might be better.

Reply:
+1 we should Errorf and pass back the error like everything else.

Reply:
@dixudx - re-ping are you going to fix this ^